### PR TITLE
[BPF] ipv6 core programs

### PR DESCRIPTION
**Pull request by tomastigera**
## Description

    - ICMPv6 is blindly allowed as (a) processing it is not implemented yet
      and (b) it performs arp's function

    - ipv6 ipsets in policy

    - bpf v6 CTLB and FIB implementation
    
    - Some code reshuffling due to stack limitations.

refs https://github.com/projectcalico/calico/issues/4736

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/bpf-gpl/connect_balancer_v46.c
Comment (fasaxc) [inline felix/bpf-gpl/connect_balancer_v46.c:37]: Worth a utility check func `is_v4_in_v6()` or similar?

File: felix/bpf/polprog/pol_prog_builder.go
Comment (fasaxc) [inline felix/bpf/polprog/pol_prog_builder.go:102]:
```suggestion
	// port+proto+pad, the dst key is also aligned in the same way. <sweet :-)>
```
Can you assert on this somehow so that we'll fail to compile if we accidentally break alignment?

File: felix/bpf/polprog/pol_prog_builder.go
Comment (fasaxc) [inline felix/bpf/polprog/pol_prog_builder.go:414]: Magic number?  Wonder if it'd be cleaner to have a couple of structs defined with the offsets for IPv4 and IPv6, then pick the right one.
```
if ipV6 {
  offsets = v6Offsets
} else {
  ofsets = v4Offsets
}

...
p.b.StoreStack8(R1, offsets.keyOffset+offsets.ipsKeyPad)
```
Or at least a comment explain the fudge factor.

File: felix/bpf/ut/pol_prog_test.go
Comment (fasaxc) [inline felix/bpf/ut/pol_prog_test.go:1618]: Do we have tests for port matches too?  Would be good to include that.

**Review by fasaxc — CHANGES_REQUESTED** [review]
Looks good, just a few small nits.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Review by fasaxc — APPROVED** [review]

**Comment by tomastigera** [comment]
An unrelated iptables flaky test failed, bpf passed, merging

**Pull request merged**
